src/date-picker/calendar/index.tsx

```diff
@@ -120,7 +133,7 @@ const Calendar = React.forwardRef(
     }, [calendarHasFocus, baseDate, isDateEnabled]);
 
     if (calendarHasFocus && !focusedDate) {
-      focusedDate = selectFocusedDate(selectedDate, baseDate);
+      selectFocusedDate(selectedDate, baseDate) && setFocusedDate(selectFocusedDate(selectedDate, baseDate));
```

Comment:
This line is a bit confusing to read, would it make sense to change it to:

```
const nextFocusedDate = selectFocusedDate(selectedDate, baseDate)
if (nextFocusedDate) {
  setFocusedDate(nextFocusedDate)
}
```

Or maybe it will be easier to just call setFocusedDate inside selectFocusedDate:

```
if (calendarHasFocus && !focusedDate) {
  selectFocusedDate(selectedDate, baseDate)
}

function selectFocusedDate(selected: Date | null, baseDate: Date): void {
   ...setFocusedDate(...)
}
```